### PR TITLE
Add Web Types to improve code assistance in WebStorm IDE and other JetBrains IDEs

### PR DESCRIPTION
**Pull request by piotrtomiak**
This contribution is to heavily improve coding experience in JetBrains IDEs (e.g. WebStorm). It contains information about components (props, events, slots) and directives. Each element is documented with a link to the documentation. With this file included, you get following experience:

element-ui-screencast 2022-09-06 11_17_09

Please make sure these boxes are checked before submitting your PR, thank you!

* [x] Make sure you follow Element's contributing guide ([中文](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.zh-CN.md) | [English](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.en-US.md) | [Español](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.fr-FR.md)).
* [x] Make sure you are merging your commits to `dev` branch.
* [x] Add some descriptions and refer relative issues for you PR.

### CODE REVIEW

**Comment by webvs2** [comment]
Wow, that sounds pretty good.

**Comment by linghengqian** [comment]
Carbon Copy @xinguanhua .

File: web-types.json [hunk 1]
```diff
@@ -0,0 +1,4195 @@
+{
```
Comment (alexlyul) [inline web-types.json:1]: Hello! It seems the package of element-ui@2.15.12 does not include this web-types.json but only it's reference from package.json ("web-types": "./web-types.json")

**Review by alexlyul — COMMENTED** [review]
IDEA hints does not work yet

File: web-types.json (hunk 1 above)
Comment (syrm-ll) [reply web-types.json:1]: <https://github.com/ElemeFE/element/pull/22271>
<https://github.com/ElemeFE/element/issues/22261>

Comment (jayzch) [reply web-types.json:1]: 求解决

**Comment by lixing0323** [comment]
弄了1天才发现element-ui 最新版本有问题，我差点怀疑人生了。各种尝试，，，才发现v2.15.11有问题，求尽快更新，要么以后就别优化ide了。

**Comment by lixing0323** [comment]
弄了1天才发现element-ui 最新版本有问题，我差点怀疑人生了。各种尝试，，，才发现v2.15.11有问题，求尽快更新，要么以后就别优化ide了。

**Comment by jayzch** [comment]
> 弄了1天才发现element-ui 最新版本有问题，我差点怀疑人生了。各种尝试，，，才发现v2.15.11有问题，求尽快更新，要么以后就别优化ide了。

闹心

File: web-types.json (hunk 1 above)
Comment (jayzch) [reply web-types.json:1]: 很遗憾，最新版本2.15.13，依旧没有解决，点击组件名称还是无法跳入组件源码

Comment (zc-write-code) [reply web-types.json:1]: > 很遗憾，最新版本2.15.13，依旧没有解决，点击组件名称还是无法跳入组件源码

对,要看源码现在只能在node_modules里面找到然后看了

Comment (nivranic) [reply web-types.json:1]: > 很遗憾，最新版本2.15.13，依旧没有解决，点击组件名称还是无法跳入组件源码
我找了几个小时，原来大家也都有这个问题，那就没事了。话说现在都6月份了，这个问题也还没解决，麻了

Comment (nivranic) [reply web-types.json:1]: > 很遗憾，最新版本2.15.13，依旧没有解决，点击组件名称还是无法跳入组件源码


我找了几个小时，原来大家也都有这个问题，那就没事了。话说现在都6月份了，这个问题也还没解决，麻了

Comment (MainPri) [reply web-types.json:1]: webstrom2023.1.4，element-ui2.15.13 还是提示Unknown html tag-xxxxx，点击组件名称无法跳转到组件源码

**Comment by young-3303** [comment]
2.15.14 版本使用Ctrl依然无法跳转到源码

**Pull request merged**